### ISSUE TITLE
[BUG] can't select first of month

### ISSUE THREAD

**Issue by devii21** [kfiross/flutter_holo_date_picker#18]
This bug was introduced with the newest version (0.3.0). On 0.2.3. it works as expected.

**To reproduce:**
  
  ```
  DatePickerWidget(
                initialDate: DateTime.now(),
                firstDate: DateTime.now(),
                onChange: (value, _) {
                  print(value.toString());
                }),
  ```
  
  -  select 2021-Jan-01
  - -> onChange value prints 2021-01-02 
      

--

This is a very critical bug so I downgraded to 0.2.3., which introduces the [old jumping date bug](https://github.com/kfiross/flutter_holo_date_picker/issues/15)

Would be highly appreciated if this could be fixed, so i can enjoy the previous bugfix 😊

**Comment by kfiross** [comment]
Hi! Sorry for the inconvenience.. I will make a fix soon

**Comment by bravelocation** [comment]
I think the problem is in line 309 in date_picker_widget.dart

```if (index == 0) return;```

If you pick the first of the month then the onSelectedChange() event is not called. At first glance, I can't see any reason for this line being there, but doing a quick test locally and commenting the line out fixed the issue for me.

**Comment by STG284** [comment]
Hey @kfiross, did you find a fix?

**Comment by kfiross** [comment]
@STG284 @devii21 @bravelocation  The issue was fixed in version 0.3.4
